extensions/librdkafka/PublishKafka.cpp

```diff
@@ -639,6 +647,17 @@ void PublishKafka::onTrigger(const std::shared_ptr<core::ProcessContext> &contex
     PublishKafka::ReadCallback callback(max_flow_seg_size, kafkaKey, thisTopic->getTopic(), conn->getConnection(), flowFile,
                                         attributeNameRegex, messages, flow_file_index);
     session->read(flowFile, &callback);
+
+    bool dropEmptyFlowFiles;
+    if (!callback.called_ && context->getProperty(DropEmptyFlowFiles.getName(), dropEmptyFlowFiles) && !dropEmptyFlowFiles) {
```

Comment:
How does it solve the original issue?
In case session->read fails due to missing content claim, it swallows exception, so the execution continues but the callback is never called? 

What's the motivation of moving the kafka related logic to the callback? 

I also have a feeling that this solution won't work in the there IS a content claim, but the content is actually empty. 
Because of this I don't think that moving the logic to the callback is a good idea.

Reply:
Producing the kafka message was part of the callback even before my changes. The issue is that when there is no `ResourceClaim`, `ProcessSession` doesn't call the callback and therefore we have no way of sending the message.

I have extended the callback to handle empty flow files and made sure it's called with the above lines. The callback will be called exactly when it was not called and the used opted in for the correct behavior by setting `DropEmptyFlowFiles` to false.

Reply:
I think the callback is the right place for producing the kafka message, since this is normally done in response to a successful read operation, as the next step of the pipeline.
I'm not sure that the workaround belongs to the callback, but since there's a lot of shared code, I think it's fine.

Reply:
Indeed, the behavior is inconsistent. When DropEmptyFlowFiles is true and there is a content claim, we don't drop the empty flow file. Will fix this.

Reply:
fixed in c872f3d